CodeModel/CodeParsers/CSharp/CaDETClassMetricCalculator.cs

```diff
@@ -69,6 +74,55 @@ private double GetLackOfCohesionOfMethods(CaDETClass parsedClass)
             return Math.Round(1 - methodFieldAccess/maxCohesion, 3);
         }
 
+        /// <summary>
+        /// LCOM - Lack of Cohesion of Methods
+        /// Article: Chidamber and Kemerer, 1991, "Towards a Metric Suite for Object-Oriented Design", Proc. OOPSLA'91, Sigplan Notices, 26(11), 197-211
+        /// </summary> 
+        private double GetLackOfCohesionOfMethods3(CaDETClass parsedClass)
+        {
+            var numberOfAttributes = GetNumberOfAttributesDefined(parsedClass);
+            var numberOfMethods = GetNumberOfMethodsDeclared(parsedClass);
+          
+            if (numberOfMethods == 0 || numberOfAttributes == 0) return 0;
+            if (numberOfMethods == 1) return 0;
+
+            double methodFieldAccess = 0;
+            foreach (var method in parsedClass.Members.Where(method => method.Type.Equals(CaDETMemberType.Method)))
+            {
+                methodFieldAccess += CountOwnFieldAndAccessorAccessed(parsedClass, method);
+            }
+            return Math.Round((numberOfMethods - methodFieldAccess/numberOfAttributes) / (numberOfMethods - 1), 3);
+        }
+
+        /// <summary>
+        /// LCOM - Lack of Cohesion of Methods
+        /// DOI: 10.1145/2723742.2723753
+        /// </summary>
+        private double GetLackOfCohesionOfMethods4(CaDETClass parsedClass)
```

Comment:
Given the complexity of this calculation we need a lot of test data. Search online if the papers don't give enough examples and have 5+ test cases for each metric.